features/behat-steps.feature

```diff
@@ -0,0 +1,741 @@
+      """
+
+  Scenario: Test "contents of file should match" regex
+    Given an empty directory
+    And a pattern.txt file:
+      """
+      Version: 1.2.3
+      """
+    Then the contents of the pattern.txt file should match /Version:\s+\d+\.\d+\.\d+/
+
+  Scenario: Test "contents of file should not match" regex
+    Given an empty directory
+    And a text.txt file:
+      """
+      No version here
+      """
+    Then the contents of the text.txt file should not match /Version:\s+\d+/
+
+  Scenario: Test "directory should contain" files
+    Given an empty directory
+    And a file1.txt file:
+      """
+      content1
+      """
+    And a file2.txt file:
+      """
+      content2
+      """
+    Then the {RUN_DIR} directory should contain:
+      """
+      file1.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file2.txt
+      """
+
+  Scenario: Test "I run the previous command again" step
+    When I run `echo "test"`
+    Then STDOUT should contain:
+      """
+      test
+      """
+
+    When I run the previous command again
+    Then STDOUT should contain:
+      """
+      test
+      """
+
+  Scenario: Test variable replacement in commands
+    When I run `echo "myvalue"`
+    Then save STDOUT as {TEST_VAR}
+
+    When I run `echo "Value is: {TEST_VAR}"`
+    Then STDOUT should contain:
+      """
+      myvalue
+      """
+
+  Scenario: Test STDOUT strictly be
+    When I run `echo "exact"`
+    Then STDOUT should strictly be:
+      """
+      exact
+      """
+
+  Scenario: Test file strictly contain
+    Given an empty directory
+    And a strict.txt file:
+      """
+      exact content
+      """
+    Then the strict.txt file should strictly contain:
+      """
+      exact content
+      """
+
+  @require-wp
+  Scenario: Test WP installation steps
+    Given a WP installation
+    When I run `wp core version`
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test WP files and wp-config.php steps
+    Given an empty directory
+    And WP files
+    Then the wp-settings.php file should exist
+
+    Given wp-config.php
+    Then the wp-config.php file should exist
+    And the wp-config.php file should contain:
+      """
+      DB_NAME
+      """
+
+  @require-wp
+  Scenario: Test WP database step
+    Given a WP installation
+    And a database
+    When I run `wp db check`
+    Then the return code should be 0
+
+  @require-wp
+  Scenario: Test WP installation in subdirectory
+    Given a WP installation in 'subdir'
+    When I run `wp core version` from 'subdir'
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test plugin installation step
+    Given a WP installation
+    And these installed and active plugins:
+      """
+      hello
+      """
+    When I run `wp plugin list --status=active --field=name`
+    Then STDOUT should contain:
+      """
+      hello
+      """
```

Comment:
Maybe need to remove this test as `wp plugin` isn't available in tests here out of the box.